File: frontend/lib/branding.ts
export interface BrandingConfig {
  companyName: string;
  productName: string;
  logoUrl: string;
  darkLogoUrl: string;
  faviconUrl: string;
  footerText: string;
  supportEmail: string;
  themeColor: string;
  termsUrl: string;
  privacyUrl: string;
  seoTitleDefault: string;
  seoDescriptionDefault: string;
}

export const BRANDING_DEFAULTS: BrandingConfig = {
  companyName: 'Caldim Engineering',
  productName: 'CAL-RIMS',
  logoUrl: '/calrims/logo.png',
  darkLogoUrl: '/calrims/logo-dark.png',
  faviconUrl: '/calrims/logo.png',
  footerText: 'Powered by Caldim Engineering. Built for teams who care about who they hire.',
  supportEmail: 'support@caldimproducts.com',
  themeColor: '#2563eb', // Default blue color
  termsUrl: '/calrims/terms/',
  privacyUrl: '/calrims/privacy/',
  seoTitleDefault: 'CAL-RIMS - AI-Powered Recruitment Intelligence System',
  seoDescriptionDefault: 'CAL-RIMS is an AI-powered automated recruitment platform for seamless hiring, empowering teams to find, evaluate, and hire top-tier talent efficiently.',
};

export function getBranding(settings: any): BrandingConfig {
  // DB has highest priority, then env variables, then default constants
  const companyName = settings?.company_name || process.env.NEXT_PUBLIC_COMPANY_NAME || BRANDING_DEFAULTS.companyName;
  const productName = settings?.product_name || process.env.NEXT_PUBLIC_PRODUCT_NAME || BRANDING_DEFAULTS.productName;
  const logoUrl = settings?.company_logo_url || process.env.NEXT_PUBLIC_LOGO_URL || BRANDING_DEFAULTS.logoUrl;
  const darkLogoUrl = settings?.dark_logo_url || process.env.NEXT_PUBLIC_DARK_LOGO_URL || BRANDING_DEFAULTS.darkLogoUrl;
  const faviconUrl = settings?.favicon_url || process.env.NEXT_PUBLIC_FAVICON_URL || BRANDING_DEFAULTS.faviconUrl;
  const footerText = settings?.footer_text || process.env.NEXT_PUBLIC_FOOTER_TEXT || BRANDING_DEFAULTS.footerText;
  const supportEmail = settings?.support_email || process.env.NEXT_PUBLIC_SUPPORT_EMAIL || BRANDING_DEFAULTS.supportEmail;
  const themeColor = settings?.theme_color || process.env.NEXT_PUBLIC_THEME_COLOR || BRANDING_DEFAULTS.themeColor;
  const termsUrl = settings?.terms_url || process.env.NEXT_PUBLIC_TERMS_URL || BRANDING_DEFAULTS.termsUrl;
  const privacyUrl = settings?.privacy_url || process.env.NEXT_PUBLIC_PRIVACY_URL || BRANDING_DEFAULTS.privacyUrl;
  const seoTitleDefault = settings?.seo_title_default || process.env.NEXT_PUBLIC_SEO_TITLE_DEFAULT || BRANDING_DEFAULTS.seoTitleDefault;
  const seoDescriptionDefault = settings?.seo_description_default || process.env.NEXT_PUBLIC_SEO_DESCRIPTION_DEFAULT || BRANDING_DEFAULTS.seoDescriptionDefault;

  return {
    companyName,
    productName,
    logoUrl,
    darkLogoUrl,
    faviconUrl,
    footerText,
    supportEmail,
    themeColor,
    termsUrl,
    privacyUrl,
    seoTitleDefault,
    seoDescriptionDefault,
  };
}

File: backend/app/core/branding.py
import os
from sqlalchemy.orm import Session
from app.domain.models import GlobalSettings

BRANDING_DEFAULTS = {
    "company_name": "Caldim Engineering",
    "product_name": "CAL-RIMS",
    "company_logo_url": "/calrims/logo.png",
    "dark_logo_url": "/calrims/logo-dark.png",
    "favicon_url": "/calrims/logo.png",
    "footer_text": "Powered by Caldim Engineering. Built for teams who care about who they hire.",
    "support_email": "support@caldimproducts.com",
    "theme_color": "#2563eb",
    "terms_url": "/calrims/terms/",
    "privacy_url": "/calrims/privacy/",
    "seo_title_default": "CAL-RIMS - AI-Powered Recruitment Intelligence System",
    "seo_description_default": "CAL-RIMS is an AI-powered automated recruitment platform for seamless hiring, empowering teams to find, evaluate, and hire top-tier talent efficiently."
}

def get_branding_value(db: Session, key: str) -> str:
    """
    Get a branding configuration value.
    Precedence:
    1. Database `global_settings` table
    2. Environment variable (uppercase version of the key)
    3. Default value constant
    """
    # 1. Try DB
    try:
        setting = db.query(GlobalSettings).filter(GlobalSettings.key == key).first()
        if setting and setting.value is not None and setting.value not in ("[UNREADABLE]", "[DECRYPTION_ERROR]"):
            return setting.value
    except Exception:
        pass

    # 2. Try Environment Variable
    env_key = key.upper()
    env_val = os.getenv(env_key)
    if env_val is not None:
        return env_val

    # 3. Fallback to hardcoded default
    return BRANDING_DEFAULTS.get(key, "")

def get_all_branding(db: Session) -> dict:
    """
    Retrieve all branding configuration values with fallback.
    """
    # Load all settings from database to minimize queries
    db_settings = {}
    try:
        settings_records = db.query(GlobalSettings).all()
        db_settings = {s.key: s.value for s in settings_records}
    except Exception:
        pass

    branding = {}
    for key, default_val in BRANDING_DEFAULTS.items():
        # Precedence: DB -> Env -> Default
        val = db_settings.get(key)
        if val in ("[UNREADABLE]", "[DECRYPTION_ERROR]"):
            val = None
        if val is None or val == "":
            val = os.getenv(key.upper())
        if val is None or val == "":
            val = default_val
        branding[key] = val

    return branding
